python/actions/clone_review_with_latest/hook/clone_review.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# :coding: utf-8
+# :copyright: Copyright (c) 2018 ftrack
+
+import logging
+import ftrack_api
+import datetime
+from ftrack_action_handler.action import BaseAction
+
+
+class CloneReview(BaseAction):
+    '''Clone review class.'''
+
+    label = 'Clone review with Latest versions'
+    identifier = 'com.ftrack.recipes.clone_review'
+    description = 'Clone review and update with latest versions available.'
+    
+    def validate_selection(self, entities):
+        '''Return True if the selection is valid.
+
+        Utility method to check *entities* validity.
+
+        '''
+        if not entities:
+            return False
+
+        if len(entities) > 1:
+            # works on one at time only.
+            return False
+        
+        entity_type, entity_id = entities[0]
+        if entity_type == 'ReviewSession':
+            return True
+
+        return False
+    
+    def discover(self, session, entities, event):
+        '''Return True if the action can be discovered.
+
+        Check if the current selection can discover this action.
+
+        '''
+        return self.validate_selection(entities)
+
+    def launch(self, session, entities, event):
+        '''Return result of running action.'''
+
+        for entity_type, entity_id in entities:
```

Comment:
Will `entities` ever contain more than one? :)

Reply:
fair point, I'll limit to 1